core/shared/src/main/scala/cats/effect/ExitCode.scala

```diff
@@ -32,7 +51,9 @@ object ExitCode {
    *   the value whose 8 least significant bits are used to construct an exit code within the
    *   valid range.
    */
-  def apply(i: Int): ExitCode = new ExitCode(i & 0xff) {}
+  def apply(i: Int): ExitCode = new ExitCode(i & 0xff)
+
+  def unapply(ec: ExitCode): Option[Int] = Some(ec.code)
```

Comment:
```suggestion
  def unapply(ec: ExitCode): Some[Int] = Some(ec.code)
```

This way we keep exhaustiveness checking

Reply:
This is how I did it at first, but mima complained actually. So I have no idea if/how we are getting exhaustive checking right now, maybe not at the bytecode level 🤔